core/search/disjunction.hpp

```diff
@@ -319,7 +504,14 @@ class small_disjunction : public doc_iterator_base, score_ctx {
       ++begin;
     }
 
-    return (doc_.value = min);
+    doc_.value = min;
+
+    if /*constexpr*/ (std::is_same<Adapter, position_score_iterator_adapter<DocIterator>>::value) {
+      hitch_remaining_iterators(target, begin);
+      this->update_attribute_range(doc_.value, itrs_);
```

Comment:
let's avoid having this logic as a part of next/seek call, use attribute+visitor instead